OpenRA.Game/Sound/Sound.cs

```diff
@@ -93,7 +94,7 @@ public void Initialize(ISoundLoader[] loaders, IReadOnlyFileSystem fileSystem)
 			this.loaders = loaders;
 			this.fileSystem = fileSystem;
 			ISoundSource LoadIntoMemory(ISoundFormat soundFormat) => soundEngine.AddSoundSourceFromMemory(
-				soundFormat.GetPCMInputStream().ReadAllBytes(), soundFormat.Channels, soundFormat.SampleBits, soundFormat.SampleRate);
+				soundFormat.GetPCMData(), soundFormat.Channels, soundFormat.SampleBits, soundFormat.SampleRate);
```

Comment:
To avoid polluting the existing classes with extra API surface here, perhaps instead the Stream could detect a `ReadAllBytes` call by checking if a call to `Read` will consume the entire Stream - when it does it can run the fast-path scenario internally.